### ISSUE TITLE
Graceful Failure.

### ISSUE THREAD

**Issue by Trevogre** [bpampuch/pdfmake#259]
I haven't dug into the code to see why it wouldn't be possible, but it would be nice to have more graceful failure that doesn't kill a whole pdf. 

Specifically I'm working on tables. And when you don't have everything matched up perfectly on one section it kills generation. Rather than generating the rest of the pdf and expressing and error in the failing section. Or setting it to ignore errors. So if a unit can't render that it just doesn't show it at all. 

Probably a pretty major architectural change but it seems like a worthwhile effort. It would just speed up the ability to use the library and throw imperfect data at it without having to prep everything. Tables are especially irritating that you can't define columns and then have it still render if you don't explicitly call out contents for each column. 

I guess you could take the position that it promotes a better implementation to no forgive errors, but is that always the priority?

**Comment by jthoenes** [comment]
You're right, it would be great. Either that we fail graceful or we reflect better what you did wrong when throwing an error. 

However, I don't think we'll find the time for it soon (at least I won't). After all - this is not ONE feature. It's a lot of small features - and small edge cases to be thought of.

**Comment by Rainbowlemon** [comment]
+1 I'd really love to see this in action. Currently working on a project with pdfmake running on a heroku instance, and the whole app crashes if someone throws a badly formatted CSV into the builder.

**Comment by upendrakolte** [comment]
Any update on this one .. as it is creating whole lot of frustration for users as even for small issues when it does not match it simple stop rendering PDF. 
I am having this error, dont know how to fix it .. as it does not what is wrong user is doing.. i found some thing with font but really clueless how to fix it based on info:
angular.js:13156 TypeError: Cannot read property 'advanceWidth' of null
    at new EmbeddedFont (embedded.js:15)
    at Function.open (font_factory.js:28)
    at PDFDocument.font (fonts.js:51)
    at FontProvider.provideFont (fontProvider.js:49)
    at textTools.js:302
    at Array.forEach (<anonymous>)
    at measure (textTools.js:281)
    at TextTools.buildInlines (textTools.js:32)
    at DocMeasure.measureLeaf (docMeasure.js:210)
    at docMeasure.js:58
